File: PYTHON/myrandom.py
from hashlib import sha256 as libsha256

def sha256(strinp):
        return libsha256(strinp.encode()).hexdigest()
class MyRandom(object):
        BASE = 10
        LENGTH = 8
        MAX_STATE = 100000
        def __init__(self,seed):
                self.__state=0
                self.__hash=sha256(seed)
        def __getState(self):
                self.__state=(self.__state+1) % MyRandom.MAX_STATE
                return self.__state
        def generate(self):
                ret = 0
                arr=[0]*MyRandom.LENGTH
                for i in range(0,64):
                        arr[i % MyRandom.LENGTH] += ord(self.__hash[i]) % MyRandom.BASE
                for i in range(0,MyRandom.LENGTH):
                        ret = ret * MyRandom.BASE + arr[i]
                self.__hash = sha256(self.__hash + str(self.__getState()) + str(ret))
                return ret

File: JAVA/MyRandom.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package mpd;

import java.security.MessageDigest;

/**
 * @author mohalisad
 */
public class MyRandom {
    private final static int BASE = 10;
    private final static int LENGTH = 8;
    private final static int MAX_STATE = 100000;

    public static String sha256_hash(String base) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(base.getBytes("UTF-8"));
            StringBuffer hexString = new StringBuffer();
            for (int i = 0; i < hash.length; i++) {
                String hex = Integer.toHexString(0xff & hash[i]);
                if (hex.length() == 1) hexString.append('0');
                hexString.append(hex);
            }
            return hexString.toString();
        } catch (Exception ex) {
            throw new RuntimeException(ex);
        }
    }

    private int state;
    private String hash;

    private int getState() {
        state = (state + 1) % MAX_STATE;
        return state;
    }

    public MyRandom(String input) {
        state = 0;
        hash = sha256_hash(input);
    }

    public int generate() {
        int ret = 0;
        int[] arr = new int[LENGTH];
        for (int i = 0; i < LENGTH; i++) {
            arr[i] = 0;
        }
        for (int i = 0; i < 64; i++) {
            arr[i % LENGTH] += hash.charAt(i) % BASE;
        }
        for (int i = 0; i < LENGTH; i++) {
            ret = ret * BASE + arr[i];
        }
        hash = sha256_hash((hash + getState()) + ret);
        return ret;
    }
}
